### PR TITLE
Native Assignment Due Dates

### PR DESCRIPTION
**Pull request by balintbartok**
refs: MBL-15023
affects: Teacher
release note: none

test plan: 
- compare with react native version. 
- open assignment
- click due date
- check multiple due dates, multiple assignees, etc

## Checklist

- [x] A11y checked
- [x] Tested on phone
- [x] Tested on tablet
- [x] Tested in dark mode
- [x] Tested in light mode

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request will not generate a release note.</td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Teacher

[MBL-15023](https://instructure.atlassian.net/browse/MBL-15023)

<details><summary>:x: Yarn lint</summary>


    
    /Users/vagrant/git/rn/Teacher/src/routing/register-screens.js
      28:8  error  'CourseDetailsSplitViewPlaceholder' is defined but never used  no-unused-vars
    
    ✖ 1 problem (1 error, 0 warnings)
    
</details>
<details><summary>:x: XCTest failed: CoreTests/AssignmentDueDateItemViewModelTests/testDueDate</summary>


    XCTAssertEqual failed: ("Due Dec 25, 2022 at 7:24 AM") is not equal to ("Due Dec 25, 2022 at 3:24 PM")
</details>
<details><summary>:x: XCTest failed: CoreTests/AssignmentDueDateItemViewModelTests/testFromUntilDates</summary>


    XCTAssertEqual failed: ("Dec 6, 2022 at 7:24 AM") is not equal to ("Dec 6, 2022 at 3:24 PM")
    XCTAssertEqual failed: ("Dec 25, 2022 at 7:24 AM") is not equal to ("Dec 25, 2022 at 3:24 PM")
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2b3b0775b9e2390a4a443608211b378bda9e458d
</p>

File: Core/Core/Assignments/AssignmentDueDates/View/AssignmentDueDateItemView.swift [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                .font(.heavy24).foregroundColor(.textDarkest)
+                .padding(16)
+            Section(
+                label: Text("For"),
+                content: { Text(model.assignee) }
+            )
+            Section(
+                label: Text("Available From"),
+                content: {
+                    if let fromEmptyAccessibility = model.fromEmptyAccessibility {
```
Comment (idevainstr) [inline Core/Core/Assignments/AssignmentDueDates/View/AssignmentDueDateItemView.swift:40]: I wonder if we could have a universal text type for handling placeholders.

**Review by idevainstr — CHANGES_REQUESTED** [review]

**Review by vargaat — CHANGES_REQUESTED** [review]
The original screen had the sections grouped for voiceover, we could restore that, the experience is way better.

File: Core/Core/Assignments/AssignmentDueDates/ViewModel/AssignmentDueDateItemViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            self.from = "--"
+            self.fromEmptyAccessibility = NSLocalizedString("No available from date set.", bundle: .core, comment: "")
+        }
+
+        if let lockAt = item.lockAt?.dateTimeString {
+            self.until = lockAt
+            self.untilEmptyAccessibility = nil
+        } else {
+            self.until = "--"
+            self.untilEmptyAccessibility = NSLocalizedString("No available until date set.", bundle: .core, comment: "")
```
[the remark this answers is not in the document]
Comment (balintbartok) [reply Core/Core/Assignments/AssignmentDueDates/ViewModel/AssignmentDueDateItemViewModel.swift:55]: this is the original message, this is what we have translations for, I think it makes sense as well, set being the verb in the sentence.

File: Core/Core/Assignments/AssignmentDueDates/View/AssignmentDueDateItemView.swift (hunk 1 above)
Comment (balintbartok) [reply Core/Core/Assignments/AssignmentDueDates/View/AssignmentDueDateItemView.swift:40]: it's not that common, to have a different accessibility for a placeholder, so I'll leave this, the next time it comes up, we can think of something.

**Review by vargaat — APPROVED** [review]
QA+1

**Pull request merged**